lib/stat_tracker.rb

```diff
@@ -1,17 +1,39 @@
 require "csv"
+require_relative "./teams_collection"
+require_relative "./games_collection"
+require_relative "./game_teams_collection"
 
 class StatTracker
   attr_reader :games,
               :teams,
               :game_teams
 
-  def initialize(file_paths)
-    @games = file_paths[:games]
-    @teams = file_paths[:teams]
-    @game_teams = file_paths[:game_teams]
+  def initialize(locations)
+    @games ||= collect_games(locations[:games])
+    @teams ||= collect_teams(locations[:teams])
+    @game_teams ||= collect_game_teams(locations[:game_teams])
   end
 
```

Comment:
We can use the "Or equals operator" to separate the responsibility of assigning state from the helper methods and keep that responsibility all in the initialize method.

Reply:
Awesome!  Looks good.